hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -254,6 +257,17 @@ public void notifySnapshotInstalled(RaftProtos.InstallSnapshotResult result,
   @Override
   public TransactionContext startTransaction(
       RaftClientRequest raftClientRequest) throws IOException {
+    if (readonlyMode.get()) {
+      TransactionContext ctxt = TransactionContext.newBuilder()
+          .setClientRequest(raftClientRequest)
+          .setStateMachine(this)
+          .setServerRole(RaftProtos.RaftPeerRole.LEADER)
+          .build();
+      ctxt.setException(new OMException("OM is running in readonly mode",
```

Comment:
This should specify that it is in this state due to an internal error. Otherwise it is not clear how the OM got into this state.